### PR TITLE
Fwf 1690

### PR DESCRIPTION
**Pull request by AbijithS-aot**
* new end point added for form submission

# Issue Tracking

JIRA: 
Issue Type: BUG/ FEATURE

# Changes

new end point added for application create
# How has the change been tested?

# Screenshots (if applicable)

# Build Success screenshot (Till a CICD pipeline is set up)

image

# Notes

### CODE REVIEW

**Review by Josephalexantony-aot — APPROVED** [review]

File: forms-flow-api-utils/src/formsflow_api_utils/services/external/formio.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 """This exposes the Formio APIs."""
 import json
 from http import HTTPStatus
+from xmlrpc.client import ResponseError
```
Comment (sreehari-aot) [inline forms-flow-api-utils/src/formsflow_api_utils/services/external/formio.py:4]: unused import

**Review by sreehari-aot — CHANGES_REQUESTED** [review]

**Review by abilpraju-aot — APPROVED** [review]

File: forms-flow-api/src/formsflow_api/resources/application.py [hunk 2]
```diff
@@ -363,3 +364,68 @@ def get():
             )
         except BusinessException as err:
             return err.error, err.status_code
+
+
+@cors_preflight("POST,OPTIONS")
+@API.route("/create-application", methods=["POST", "OPTIONS"])
```
Comment (sumesh-aot) [inline forms-flow-api/src/formsflow_api/resources/application.py:370]: This is not an acceptable rest endpoint, as `/application/create` is already there, and `/application/create-application` doesn't follow rest standard.
May be make it as `/application/external/create` or something to align with other endpoint pattern we are following

File: forms-flow-api/docker-compose.yml [hunk 3]
```diff
@@ -43,6 +43,7 @@ services:
       KEYCLOAK_URL: ${KEYCLOAK_URL}
       KEYCLOAK_URL_REALM: ${KEYCLOAK_URL_REALM:-forms-flow-ai}
       WEB_API_BASE_URL: ${FORMSFLOW_API_URL}
+      WEB_BASE_URL: ${WEB_BASE_URL}
```
Comment (sumesh-aot) [inline forms-flow-api/docker-compose.yml:46]: Can you add a default null value here to break existing applications ?

File: forms-flow-api/src/formsflow_api/resources/application.py [hunk 4]
```diff
@@ -363,3 +364,68 @@ def get():
...
+        : data: form submission data
+        : formId:- Unique Id for the corresponding form
+        """
+        formio_url = current_app.config.get("FORMIO_URL")
+        web_url = current_app.config.get("WEB_BASE_URL")
+        application_json = request.get_json()
+        data = request.get_json()
+        try:
+            application_schema = ApplicationSchema()
+            dict_data = application_schema.load(application_json)
```
Comment (sumesh-aot) [inline forms-flow-api/src/formsflow_api/resources/application.py:389]: Please provide meaningful name instead of dict_data, like is it application_data ?

**Review by sumesh-aot — CHANGES_REQUESTED** [review]
Test cases are missing.

**Comment by sumesh-aot** [comment]
@AbijithS-aot Can you please make the requested changes and update the PR ?

**Comment by AbijithS-aot** [comment]
will update test case in another pr

**Comment by sumesh-aot** [comment]
@AbijithS-aot Can you change the target repo to release/5.0.1 ?

File: forms-flow-api/docker-compose.yml [hunk 5]
```diff
@@ -43,6 +43,7 @@ services:
       KEYCLOAK_URL: ${KEYCLOAK_URL}
       KEYCLOAK_URL_REALM: ${KEYCLOAK_URL_REALM:-forms-flow-ai}
       WEB_API_BASE_URL: ${FORMSFLOW_API_URL}
+      WEB_BASE_URL: ${WEB_BASE_URL:-null}
```
Comment (auslin-aot) [inline forms-flow-api/docker-compose.yml:46]: can you pls add WEB_BASE_URL in sample.env

**Pull request closed without merging**